src/core/src/op/constant.cpp

```diff
@@ -61,524 +84,233 @@ ov::op::v0::Constant::Constant(const std::shared_ptr<ngraph::runtime::Tensor>& t
 }
 OPENVINO_SUPPRESS_DEPRECATED_END
 
-ov::op::v0::Constant::Constant(const ov::Tensor& tensor) {
-    m_element_type = tensor.get_element_type();
-    m_shape = tensor.get_shape();
-    // Share data from ov::Tensor
-    m_data = std::make_shared<ov::SharedBuffer<ov::Tensor>>(static_cast<char*>(tensor.data()),
-                                                            tensor.get_byte_size(),
-                                                            tensor);
-
+Constant::Constant(const Tensor& tensor)
+    : m_element_type{tensor.get_element_type()},
+      m_shape{tensor.get_shape()},
+      m_data{
+          std::make_shared<SharedBuffer<Tensor>>(static_cast<char*>(tensor.data()), tensor.get_byte_size(), tensor)} {
     constructor_validate_and_infer_types();
 }
 
-ov::op::v0::Constant::Constant(const element::Type& type,
-                               const ov::Shape& shape,
-                               const std::vector<std::string>& values)
-    : Constant(false, type, shape) {
-    NGRAPH_SUPPRESS_DEPRECATED_START
-    NODE_VALIDATION_CHECK(this,
-                          values.size() == shape_size(m_shape) || values.size() == 1,
-                          "Did not get the expected number of literals for a constant of shape ",
-                          m_shape,
-                          " (got ",
-                          values.size(),
-                          ", expected ",
-                          shape_size(m_shape),
-                          ".");
-
-    using Type_t = element::Type_t;
-
-    if (values.size() == 1 && shape_size(m_shape) != 1) {
-        // broadcast single value
-        switch (m_element_type) {
-        case Type_t::boolean:
-            fill_data<Type_t::boolean>(stoi(values[0]));
-            break;
-        case Type_t::bf16:
-            fill_data<Type_t::bf16>(ngraph::parse_string<float>(values[0]));
-            break;
-        case Type_t::f16:
-            fill_data<Type_t::f16>(ngraph::parse_string<float>(values[0]));
-            break;
-        case Type_t::f32:
-            fill_data<Type_t::f32>(ngraph::parse_string<float>(values[0]));
-            break;
-        case Type_t::f64:
-            fill_data<Type_t::f64>(ngraph::parse_string<double>(values[0]));
-            break;
-        case Type_t::i4:
-            fill_data<Type_t::i4>(ngraph::parse_string<int64_t>(values[0]));
-            break;
-        case Type_t::i8:
-            fill_data<Type_t::i8>(ngraph::parse_string<int64_t>(values[0]));
-            break;
-        case Type_t::i16:
-            fill_data<Type_t::i16>(ngraph::parse_string<int64_t>(values[0]));
-            break;
-        case Type_t::i32:
-            fill_data<Type_t::i32>(ngraph::parse_string<int64_t>(values[0]));
-            break;
-        case Type_t::i64:
-            fill_data<Type_t::i64>(ngraph::parse_string<int64_t>(values[0]));
```

Comment:
@praasz could you elaborate how int64 would fit in double? looks like we've lost accuracy here. Please fix it.
cc @slyalin

Reply:
Ticket created and scheduled [CVS-124982](https://jira.devtools.intel.com/browse/CVS-124982).